### PR TITLE
SNMP: Add tests for Plugin class

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [ ] ~Do all end-to-end tests pass?~
* [x] ~Any other testing performed?~
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        ip=TARGET_IP,
+        operating_system=OperatingSystem.LINUX,
+        ports_status=open_snmp_ports,
+    )
+
+
+@pytest.fixture
+def propagation_credentials_repository():
+    return MagicMock(spec=IPropagationCredentialsRepository)
+
```
Comment (ilija-lazoroski) [inline monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py:48]: We can live without this fixture.

File: monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+def test_run__skips_exploit_if_port_status_closed(
+    plugin: Plugin,
+    mock_snmp_exploiter,
+    target_host: TargetHost,
+):
+    host = target_host
+    host.ports_status.udp_ports = PortScanDataDict(
+        {
+            SNMP_PORTS[0]: PortScanData(port=SNMP_PORTS[0], status=PortStatus.CLOSED),
```
Comment (ilija-lazoroski) [inline monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py:143]: NIT: Maybe add another port as all ports should be closed

File: monkey/agent_plugins/exploiters/snmp/src/plugin.py [hunk 3]
```diff
@@ -55,7 +54,6 @@ def __init__(
         agent_binary_repository: IAgentBinaryRepository,
         http_agent_binary_server_registrar: IHTTPAgentBinaryServerRegistrar,
         propagation_credentials_repository: IPropagationCredentialsRepository,
-        tcp_port_selector: ITCPPortSelector,
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/snmp/src/plugin.py:58]: This is part of the plugin interface. Plugins should adhere to the plugin interface. "Explicit is better than implicit."

The role of `plugin.py` is to transform the Plugin Interface into the specific plugin's dependencies. This is why `plugin.py` deserializes options into a pydantic object, or uses `partial` to create callbacks for the plugin. Each plugin is unique and has its own set of requirements/dependencies that are derived from the total set of dependencies provided by the agent. `plugin.py` is the bridge between these two things.

File: monkey/agent_plugins/exploiters/snmp/src/plugin.py [hunk 4]
```diff
@@ -66,7 +64,6 @@ def __init__(
         self._agent_binary_repository = agent_binary_repository
         self._http_agent_binary_server_registrar = http_agent_binary_server_registrar
         self._propagation_credentials_repository = propagation_credentials_repository
-        self._tcp_port_selector = tcp_port_selector
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/snmp/src/plugin.py:69]: This is fine, though. No reason to save it off if we're not going to use it.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        {SNMP_PORTS[0]: PortScanData(port=SNMP_PORTS[0], status=PortStatus.OPEN)}
+    )
+    result = plugin.run(
+        host=host,
+        options={},
+        interrupt=Event(),
+    )
+
+    mock_snmp_exploiter.exploit_host.assert_called_once()  # type: ignore [attr-defined]
+    assert result == EXPLOITER_RESULT
```
Comment (mssalvatore) [inline monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py:127]: NIT: You could parametrize these. I'm fine with them being separate, though. Either way.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**